tests/tx/scripts/test_sighash.py

```diff
@@ -0,0 +1,321 @@
+        with pytest.raises(InputOutputMismatch):
+            self.manager1.verification_service.verify(atomic_swap_tx)
+
+        # Alice sends the tx bytes to Bob, represented here by cloning the tx
+        atomic_swap_tx_clone = cast(Transaction, atomic_swap_tx.clone())
+        self.manager1.cpu_mining_service.resolve(atomic_swap_tx_clone)
+
+        # Bob creates an input spending all genesis HTR and adds it to the atomic swap tx
+        htr_input = TxInput(not_none(self.genesis_block.hash), 0, b'')
+        atomic_swap_tx_clone.inputs.append(htr_input)
+
+        # Bob adds an output to receive all test tokens
+        bob_output_script = P2PKH.create_output_script(self.address2)
+        tokens_output = TxOutput(token_creation_utxo.value, bob_output_script, 1)
+        atomic_swap_tx_clone.outputs.append(tokens_output)
+
+        # Bob adds a change output for his HTR
+        htr_output = TxOutput(int(genesis_utxo.value / 2), bob_output_script)
+        atomic_swap_tx_clone.outputs.append(htr_output)
+
+        # Bob signs his input using sighash_all to complete the tx
+        data_to_sign2 = atomic_swap_tx_clone.get_sighash_all()
+        assert self.manager2.wallet
+        public_bytes2, signature2 = self.manager2.wallet.get_input_aux_data(data_to_sign2, self.genesis_private_key)
+        htr_input.data = P2PKH.create_input_data(public_bytes2, signature2)
+
+        # The atomic swap tx is now completed and valid, and can be propagated
+        self.manager1.verification_service.verify(atomic_swap_tx_clone)
+        self.manager1.propagate_tx(atomic_swap_tx_clone, fails_silently=False)
+
+    @patch('hathor.transaction.scripts.opcode.is_opcode_valid', lambda _: True)
+    def test_sighash_bitmask_with_limit(self) -> None:
+        # Create a new test token
+        token_creation_tx = create_tokens(self.manager1, self.address1_b58)
+        token_uid = token_creation_tx.tokens[0]
+        token_creation_utxo = token_creation_tx.outputs[0]
+        genesis_utxo = self.genesis_block.outputs[0]
+        parents = self.manager1.get_new_tx_parents()
+
+        # Alice creates an input spending all created test tokens
+        tokens_input = TxInput(not_none(token_creation_tx.hash), 0, b'')
+
+        # Alice creates an output sending half genesis HTR to herself
+        alice_output_script = P2PKH.create_output_script(self.address1)
+        htr_output = TxOutput(int(genesis_utxo.value / 2), alice_output_script)
+
+        # Alice creates an atomic swap tx that's missing only Bob's input, with half genesis HTR
+        atomic_swap_tx = Transaction(
+            weight=1,
+            inputs=[tokens_input],
+            outputs=[htr_output],
+            parents=parents,
+            tokens=[token_uid],
+            storage=self.manager1.tx_storage,
+            timestamp=token_creation_tx.timestamp + 1
+        )
+        self.manager1.cpu_mining_service.resolve(atomic_swap_tx)
```

Comment:
Same here.

Reply:
the tx is verified on line 172, so it needs to be resolved